src/components/utils.ts

```diff
@@ -607,6 +607,21 @@ export function getValidUrl(url: string): string {
   }
 }
 
+/**
+ * Creates a UUID v4 version
+ *
+ * @returns {string}
+ */
+export function generateUuidv4(): string {
```

Comment:
I think it would be better to name this method just as `generateId()` and make `UUIDv4` "private". E.g.
```js
export function generateId(): string {
    return uuidv4();
}

function uuidv4(): string {
  // your code
}
```
so that we could change the id generator algorithm